Subject: [PATCH] fix
fix
Remove linux sources
Fix for porting this repo in vcpkg, and use as a dependency of ffmpeg.
---
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	(revision 7e4d221c36c630c1250b23a5dfa15657bc04c10c)
+++ b/CMakeLists.txt	(revision 359299f6c49cc45f0065e9242189b2b3654e1c9c)
@@ -3,16 +3,7 @@
 project( libmfx )

 # FIXME Adds support for using system/other install of intel media sdk
-find_path ( INTELMEDIASDK_PATH mfx/mfxvideo.h
-  HINTS "${CMAKE_SOURCE_DIR}"
-)
-
-if (INTELMEDIASDK_PATH_NOTFOUND)
-  message( FATAL_ERROR "Intel MEDIA SDK include not found" )
-else (INTELMEDIASDK_PATH_NOTFOUND)
-  message(STATUS "Intel Media SDK is here: ${INTELMEDIASDK_PATH}")
-endif (INTELMEDIASDK_PATH_NOTFOUND)
-
+set(INTELMEDIASDK_PATH "${CMAKE_CURRENT_LIST_DIR}")

 set(SOURCES
   src/main.cpp
@@ -40,6 +26,10 @@
     src/mfx_load_plugin.cpp
     src/mfx_plugin_hive.cpp
     src/mfx_win_reg_key.cpp
+
+    # If not add this file, a ffmpeg build test will not pass due to missing symbols in this file
+    # (Building ffmpeg in vcpkg use MSVC toolchain can reproduce)
+    src/mfx_driver_store_loader.cpp
   )
 endif (CMAKE_SYSTEM_NAME MATCHES "Windows")

@@ -53,9 +43,24 @@
   -D_LIB
 )

-configure_file (${CMAKE_SOURCE_DIR}/libmfx.pc.cmake ${CMAKE_BINARY_DIR}/libmfx.pc @ONLY)
+# As this library might be used by ffmpeg, and in vcpkg build pipeline, the .a suffix is not recognized by the linker,
+# thus, we need to rename to .lib. (MSVC toolchain should all need this)
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+  configure_file (${CMAKE_SOURCE_DIR}/libmfx.pc.win.cmake ${CMAKE_BINARY_DIR}/libmfx.pc @ONLY)
+else()
+  configure_file (${CMAKE_SOURCE_DIR}/libmfx.pc.cmake ${CMAKE_BINARY_DIR}/libmfx.pc @ONLY)
+endif()
+

 add_library( mfx STATIC ${SOURCES} )
+
+# As this library might be used by ffmpeg, and in vcpkg build pipeline, mfx.lib won't be recognized by linker,
+# thus, we need to rename to libmfx.lib
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+  set_target_properties(mfx
+          PROPERTIES PREFIX lib)
+endif()
+
 install (DIRECTORY ${CMAKE_SOURCE_DIR}/mfx DESTINATION ${CMAKE_INSTALL_PREFIX}/include FILES_MATCHING PATTERN "*.h")
 install (FILES ${CMAKE_BINARY_DIR}/libmfx.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
 install (TARGETS mfx ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
Index: libmfx.pc.win.cmake
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/libmfx.pc.win.cmake b/libmfx.pc.win.cmake
new file mode 100644
--- /dev/null	(revision f17d65c93a28f40cedf4aa266883ca4b59f40eac)
+++ b/libmfx.pc.win.cmake	(revision f17d65c93a28f40cedf4aa266883ca4b59f40eac)
@@ -0,0 +1,14 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: libmfx
+Description: Intel Media SDK Dispatched static library
+Version: 2013
+Requires:
+Requires.private:
+Conflicts:
+Libs: -L${libdir} -lsupc++ ${libdir}/libmfx.lib
+Libs.private:
+Cflags: -I${includedir}
Index: src/mfx_driver_store_loader.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/mfx_driver_store_loader.cpp b/src/mfx_driver_store_loader.cpp
--- a/src/mfx_driver_store_loader.cpp	(revision 7e4d221c36c630c1250b23a5dfa15657bc04c10c)
+++ b/src/mfx_driver_store_loader.cpp	(revision 6a863c1a050359a3b82ec290a202feddb3e16fbd)
@@ -24,6 +24,10 @@
 #include "mfx_dispatcher_log.h"
 #include "mfx_load_dll.h"

+// If not add this, building ffmpeg in vcpkg will fail missing StringFromGUID2 symbol
+// As the ffmpeg building doesn't link Ole32
+#pragma comment(lib, "Ole32.lib")
+
 namespace MFX
 {

Index: libmfx.pc.cmake
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/libmfx.pc.cmake b/libmfx.pc.cmake
--- a/libmfx.pc.cmake	(revision e729d11bfdc0d747bc82c3f4cdef677f2f6d420c)
+++ b/libmfx.pc.cmake	(revision 359299f6c49cc45f0065e9242189b2b3654e1c9c)
@@ -11,4 +11,4 @@
 Conflicts:
 Libs: -L${libdir} -lsupc++ ${libdir}/libmfx.a
 Libs.private:
-Cflags: -I${includedir} -I@INTELMEDIASDK_PATH@
+Cflags: -I${includedir}
